src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ProcessLinkerXmlBase.cs

```diff
@@ -182,90 +188,47 @@ protected virtual void ProcessTypes(AssemblyDefinition assembly, XPathNavigator
                         continue;
                 }
 
-                TypeDefinition type = assembly.MainModule.GetType(fullname);
-
-                if (type == null && assembly.MainModule.HasExportedTypes)
-                {
-                    foreach (var exported in assembly.MainModule.ExportedTypes)
-                    {
-                        if (fullname == exported.FullName)
-                        {
-                            var resolvedExternal = ProcessExportedType(exported, assembly, typeNav);
-                            if (resolvedExternal != null)
-                            {
-                                type = resolvedExternal;
-                                break;
-                            }
-                        }
-                    }
-                }
+                TypeDesc type = CustomAttributeTypeNameParser.GetTypeByCustomAttributeTypeName(assembly, fullname, throwIfNotFound: false);
 
                 if (type == null)
                 {
-                    if (warnOnUnresolvedTypes)
-                        LogWarning(typeNav, DiagnosticId.XmlCouldNotResolveType, fullname);
+                    //if (warnOnUnresolvedTypes)
+                    //    LogWarning(typeNav, DiagnosticId.XmlCouldNotResolveType, fullname);
                     continue;
                 }
 
                 ProcessType(type, typeNav);
             }
         }
 
-        protected virtual TypeDefinition? ProcessExportedType(ExportedType exported, AssemblyDefinition assembly, XPathNavigator nav) => exported.Resolve();
-
-        void MatchType(TypeDefinition type, Regex regex, XPathNavigator nav)
+        void MatchType(TypeDesc type, Regex regex, XPathNavigator nav)
         {
-            if (regex.Match(type.FullName).Success)
+            if (regex.Match(type.GetDisplayName()).Success)
```

Comment:
Use the CecilTypeNameFormatter to format a name instead - there's differences for nested types.